protocols/migration/migration_reports_5_0_0_to_reports_4_0_0.py

```diff
@@ -0,0 +1,175 @@
+
+        # references has been renamed to supportingEvidence
+        new_instance.supportingEvidence = old_instance.references
+
+        # converts all reported variants
+        if old_instance.variants:
+            new_instance.candidateVariants = [
+                self.migrate_reported_variant(old_reported_variant=old_variant) for old_variant in old_instance.variants
+            ]
+
+        if old_instance.additionalAnalysisPanels is not None:
+            new_instance.additionalAnalysisPanels = [
+                self.migrate_analysis_panel(old_panel=panel) for panel in old_instance.additionalAnalysisPanels
+            ]
+
+        return self.validate_object(object_to_validate=new_instance, object_type=self.new_model.ClinicalReportRD)
+
+    def migrate_analysis_panel(self, old_panel):
+        new_panel = self.new_model.AdditionalAnalysisPanel(
+            panelVersion=old_panel.panel.panelVersion,
+            panelName=old_panel.panel.panelName,
+            specificDisease=old_panel.specificDisease,
+        )
+        return self.validate_object(object_to_validate=new_panel, object_type=self.new_model.AdditionalAnalysisPanel)
+
+    def migrate_reported_variant(self, old_reported_variant):
+        new_reported_variant = self.new_model.ReportedVariant(
+            dbSnpId=old_reported_variant.dbSnpId,
+            chromosome=old_reported_variant.variantCoordinates.chromosome,
+            position=old_reported_variant.variantCoordinates.position,
+            reference=old_reported_variant.variantCoordinates.reference,
+            alternate=old_reported_variant.variantCoordinates.alternate,
+            additionalTextualVariantAnnotations=old_reported_variant.additionalTextualVariantAnnotations,
+            evidenceIds=old_reported_variant.references,
+            comments=old_reported_variant.comments,
+        )
+
+        new_reported_variant.calledGenotypes = self.migrate_variant_calls_to_called_genotypes(old_reported_variant.variantCalls)
+
+        new_reported_variant.reportEvents = self.migrate_report_events(old_reported_variant.reportEvents)
+
+        new_reported_variant.additionalNumericVariantAnnotations = self.merge_annotations_and_frequencies(
+            old_reported_variant.additionalNumericVariantAnnotations, old_reported_variant.alleleFrequencies,
+        )
+        return self.validate_object(object_to_validate=new_reported_variant, object_type=self.new_model.ReportedVariant)
+
+    @staticmethod
+    def merge_annotations_and_frequencies(numeric_annotations, allele_frequencies):
+        if not isinstance(numeric_annotations, dict):
+            raise MigrationError("additionalNumericVariantAnnotations should be dict but is: {}".format(numeric_annotations))
+        if allele_frequencies is not None:
+            for af in allele_frequencies:
+                if numeric_annotations.get("{}:{}".format(af.study, af.population)) is not None:
```

Comment:
why not just `if "{}:{}".format(af.study, af.population) in numeric_annotations:` ?

Reply:
It's the same logic

Reply:
indeed, but I incline for this for two reasons:
1. Readability
2. Conciseness

Actually the most readable will be
```
annotation_key = "{}:{}".format(af.study, af.population)
if annotation_key in numeric_annotations:
   logging.warning(
                        "{} already exists in numeric_annotations with value {} instead of {}".format(
                            annotation_key, numeric_annotations.get(annotation_key), af.alternateFrequency
                        )
                    )
```

Reply:
I think we're getting into the realms of style here... personally I don't see the difference but if its the only thing that'll get it through review I'll change it to your way :stuck_out_tongue: